tests/infer/mcmc/test_hmc.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import absolute_import, division, print_function
+
+import logging
+from collections import defaultdict, namedtuple
+
+import os
+import pytest
+import torch
+from torch.autograd import Variable
+
+import pyro
+import pyro.distributions as dist
+from pyro.distributions.util import torch_ones_like, torch_zeros_like
+from pyro.infer.mcmc.hmc import HMC
+from pyro.infer.mcmc.mcmc import MCMC
+from tests.common import assert_equal
+
+logging.basicConfig(format='%(levelname)s %(message)s')
+logging.getLogger('pyro').setLevel(logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class GaussianChain(object):
+
+    def __init__(self, dim, chain_len, num_obs):
+        self.dim = dim
+        self.chain_len = chain_len
+        self.num_obs = num_obs
+        self.mu_0 = Variable(torch_zeros_like(torch.Tensor(self.dim)), requires_grad=True)
+        self.lambda_prec = Variable(torch_ones_like(torch.Tensor(self.dim)))
+
+    def model(self, data):
+        mu = pyro.param('mu_0', self.mu_0)
+        lambda_prec = self.lambda_prec
+        for i in range(1, self.chain_len + 1):
+            mu = pyro.sample('mu_{}'.format(i), dist.normal, mu=mu, sigma=Variable(lambda_prec.data))
+        pyro.sample('obs', dist.normal, mu=mu, sigma=Variable(lambda_prec.data), obs=data)
+
+    @property
+    def data(self):
+        return Variable(torch_ones_like(torch.Tensor(self.num_obs, self.dim)))
+
+    def id_fn(self):
+        return 'dim={}_chain-len={}_num_obs={}'.format(self.dim, self.chain_len, self.num_obs)
+
+
+def rmse(t1, t2):
+    return (t1 - t2).pow(2).mean().sqrt()
+
+
+T = namedtuple('TestExample', [
+    'fixture',
+    'num_samples',
+    'warmup_steps',
+    'hmc_params',
+    'expected_means',
+    'expected_precs',
+    'mean_tol',
+    'std_tol'])
+
+TEST_CASES = [
+    T(
+        GaussianChain(dim=10, chain_len=3, num_obs=1),
+        num_samples=800,
+        warmup_steps=50,
+        hmc_params={'step_size': 0.5,
+                    'num_steps': 4},
+        expected_means=[0.25, 0.50, 0.75],
+        expected_precs=[1.33, 1, 1.33],
+        mean_tol=0.04,
+        std_tol=0.06,
+    ),
+    T(
+        GaussianChain(dim=10, chain_len=4, num_obs=1),
+        num_samples=1200,
+        warmup_steps=300,
+        hmc_params={'step_size': 0.46,
+                    'num_steps': 5},
+        expected_means=[0.20, 0.40, 0.60, 0.80],
+        expected_precs=[1.25, 0.83, 0.83, 1.25],
+        mean_tol=0.06,
+        std_tol=0.06,
+    ),
+    # XXX: Very sensitive to HMC parameters. Biased estimate is obtained
```

Comment:
Are the other tests run during integration tests?

Reply:
They are not. I have only kept them so that they can be run locally, if needed. The third one was used for profiling, for instance.